core/repository/referredUsers/Node.ts

```diff
@@ -0,0 +1,60 @@
+import { Account, Address } from 'shared/model/account';
+import config from 'shared/config';
+
+export enum FactorType {
+    COUNT = 'count',
+    REWARD = 'reward',
+    STAKE_AMOUNT = 'stakeAmount'
+}
+
+export type Factors = Map<FactorType, number>;
+
+export default class Node {
+
+    readonly address: Address;
+
+    parent: Node;
+
+    children: Map<Address, Node>;
+
+    factors: Array<Factors>;
```

Comment:
how about rename it to something more valuable. like `levelStatistic` or something like that?